internal/exec/exec.go

```diff
@@ -0,0 +1,514 @@
+// Package exec implements a minimalized process execution wrapper.
+package exec
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	errProcNotStarted  = errors.New("process has not started yet")
+	errProcNotFinished = errors.New("process has not finished yet")
+)
+
+type Exec struct {
+	path         string
+	cmdline      string
+	process      *os.Process
+	procState    *os.ProcessState
+	waitCalled   bool
+	stdioOurEnd  [3]*os.File
+	stdioProcEnd [3]*os.File
```

Comment:
I feel like these names are confusing. Maybe `stdioProcEnd` is fine, but `stdioOurEnd` is a little iffy. Maybe add some comments here about what these are and see if there's another name we could use instead of `stdioOurEnd`

Reply:
Would adding 'pipes' to the name and changing End to side ease the confusion? eg. `stdioPipesOurSide` and `stdioPipesProcSide`? (in addition to adding the comment)

Reply:
Changed to the above, let me know if this is sufficient